app/components/trainees/record_details/view.rb

```diff
@@ -39,6 +39,21 @@ def trn_row
         end
       end
 
+      def trainee_status_row
+        {
+          key: "Trainee status",
+          value: render(StatusTag::View.new(trainee: trainee, css_classes: "govuk-!-margin-bottom-2")) + tag.br + trainee_status_date,
+        }
+      end
+
+      def trainee_status_date
+        status_date = trainee.deferred? ? trainee.defer_date : trainee.withdraw_date
+        status_date_prefix = trainee.deferred? ? "Deferral date: " : "Withdrawal date: "
+        unless status_date.nil?
+          status_date_prefix + date_for_summary_view(status_date)
+        end
+      end
```

Comment:
I would prefer to return early in situations like this, if there's no `status_date` then you also don't need to go ahead and calculate the `status_date_prefix`:

```suggestion
      def trainee_status_date
        status_date = trainee.deferred? ? trainee.defer_date : trainee.withdraw_date
        return unless status_date
        
        status_date_prefix = trainee.deferred? ? "Deferral date: " : "Withdrawal date: "
        status_date_prefix + date_for_summary_view(status_date)
      end
```

Reply:
Thanks! I was trying to figure out the best way to write this for ages.